### ISSUE TITLE
Save Member resets password

### ISSUE THREAD

**Issue by Myster** [umbraco/Umbraco-CMS#6283]
When I save a member in the back office their password is re-set.

Reproduction
------------
Navigate to a member, open member, press save.
The green pannel appears in the Enter your password section saying the password has been reset.
![image](https://user-images.githubusercontent.com/354777/64294503-16d2d900-cfc3-11e9-91ab-43ac7426f506.png)
I don't have any MemberService event handlers.

Note: I tried to reproduce this in an unmodified Umbraco instance, and I see a different issue.
The second time I save a member I get a validation error that the password may not be blank. which is also bad.

### Specifics
URL: http://localhost:7122/umbraco#/member/member/edit/6825f768-5527-4cbc-8ea9-f0cdbbeb1158
Umbraco version 7.15.1 assembly: 1.0.7144.21656
Browser: Chrome 76
Logging on debug:
```
2019-09-05 10:00:00,124 [P84168/D3/T27] DEBUG Umbraco.Web.UmbracoModule - Begin request: http://localhost:7122/umbraco/backoffice/UmbracoApi/Member/PostSave.
2019-09-05 10:00:00,201 [P84168/D3/T27] DEBUG Umbraco.Core.Sync.ServerMessengerBase - Invoking refresher Umbraco.Web.Cache.MemberCacheRefresher on local server for message type RefreshByInstance
2019-09-05 10:00:00,205 [P84168/D3/T27] DEBUG UmbracoExamine.DataServices.UmbracoLogService - ReIndexNode with type: member, Provider=InternalMemberIndexer, NodeId=1424
2019-09-05 10:00:00,205 [P84168/D3/T27] DEBUG UmbracoExamine.DataServices.UmbracoLogService - AddSingleNodeToIndex with type: member, Provider=InternalMemberIndexer, NodeId=1424
2019-09-05 10:00:00,207 [P84168/D3/T75] DEBUG UmbracoExamine.DataServices.UmbracoLogService - Index created for node 1424, Provider=InternalMemberIndexer, NodeId=1424
2019-09-05 10:00:00,224 [P84168/D3/T27] DEBUG Umbraco.Core.Sync.ServerMessengerBase - Invoking refresher Umbraco.Web.Cache.MemberCacheRefresher on local server for message type RefreshByInstance
2019-09-05 10:00:00,230 [P84168/D3/T27] DEBUG UmbracoExamine.DataServices.UmbracoLogService - ReIndexNode with type: member, Provider=InternalMemberIndexer, NodeId=1424
2019-09-05 10:00:00,230 [P84168/D3/T27] DEBUG UmbracoExamine.DataServices.UmbracoLogService - AddSingleNodeToIndex with type: member, Provider=InternalMemberIndexer, NodeId=1424
2019-09-05 10:00:00,231 [P84168/D3/T74] DEBUG UmbracoExamine.DataServices.UmbracoLogService - Index created for node 1424, Provider=InternalMemberIndexer, NodeId=1424
```

### Expected result
The password should not change.
### Actual result
The password was changed to a random password.

---
_This item has been added to our backlog [AB#2538](https://umbraco.visualstudio.com/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/2538)_

**Comment by nul800sebastiaan** [comment]
Yikes! I can reproduce (similar) behavior in 7.15.3 when `allowManuallyChangingPassword` is set to `true`. We'll get it fixed! 

For now it might be an option to set `allowManuallyChangingPassword` to `false` if you have a "forgot password" page for your members.

**Comment by nul800sebastiaan** [comment]
Weirdly: the first time you save a member, it all goes well. Then, only the second time you save them, the password gets reset. We somehow send: 

```json
	"password": {
		"newPassword": null,
		"reset": true
	},
```

To the server after the first save. Reloading the member after saving them helps, at that point we send: 

```json
"password": null,
```

So we must be wrongly populating that password property after the first submit.

**Comment by nul800sebastiaan** [comment]
Upon further investigation: `hasPassword` is set to `true` when we first load the member but when we get the member model back after saving the member it is set to `false` - that is where the real error comes from.

**Comment by nul800sebastiaan** [comment]
Which leads to this line of code, we set `hasPassword` to `false` when the raw password value is null

https://github.com/umbraco/Umbraco-CMS/blob/v7/dev/src/Umbraco.Web/Models/Mapping/MemberModelMapper.cs#L372

And.. yeah, that's what we do, on purpose so that we don't save it:

**Comment by nul800sebastiaan** [comment]
That wasn't it either, I think I finally found it: 

https://github.com/umbraco/Umbraco-CMS/blob/v7/dev/src/Umbraco.Web.UI.Client/src/views/member/member.edit.controller.js#L136

There's a missing check to see if `passwordProp.value.reset` is undefined, as the last check (`!passwordProp.value.reset`) returns `true` if `passwordProp.value.reset` is undefined, this is incorrect and will set `passwordProp.value.reset` to true.

**Comment by Myster** [comment]
Thanks for sorting this out,  i’m glad I’m not the only one. :)

**Comment by elit0451** [comment]
The issue has been resolved in v7 and the fix for v8 is on the way :slightly_smiling_face:

**Comment by robertjf** [comment]
Is there an eta when we can expect this to be released into the wild?  it's been 3 months now and this issue is affecting us big time.

**Comment by paulnewzealand** [comment]
Hi, this issue is closed and it says fixed in 7.5.14 but that isn't released yet - do you know when it will be?
